### PR TITLE
Add ImageSet creation documentation [pick #4421]

### PR DESCRIPTION
**Pull request by tmjd**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]

File: maintenance/image-options/imageset.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+Versioned container images are typically referenced by a tag which is appended to an image reference. Example: `<repo>/<image>:<tag>`. Container image tags are typically not expected be changed or updated, but this is not required or enforced by most image registries, meaning it is possible to push new code to the same image tag.
+
+#### Image digest
+
+Container images, when added to a container registry, have a unique hash created that can be used to pull a specific version of an image that cannot be changed or updated.
+
+### Before you begin
+
+**Required**
+- Docker client is configured to pull images from the container registries where images are stored.
```
Comment (caseydavenport) [inline maintenance/image-options/imageset.md:41]: We should probably say this requires installation using the operator

Comment (tmjd) [reply maintenance/image-options/imageset.md:41]: That info is in the "Big picture" but are you suggesting to include it here also?

Comment (tmjd) [reply maintenance/image-options/imageset.md:41]: I created https://github.com/projectcalico/calico/pull/4437 with this change. After it is merged I'll pick that change to this PR also.

Comment (caseydavenport) [reply maintenance/image-options/imageset.md:41]: Yeah, I missed that in the big picture. I think it's worth calling out in the "requirements" section.

Comment (tmjd) [reply maintenance/image-options/imageset.md:42]: Updated

**Pull request merged**